app/src/main/java/com/mstudio/superheroarch/MainActivity.kt

```diff
@@ -40,19 +43,26 @@ class MainActivity : AppCompatActivity() {
         call.enqueue(object : Callback<CharactersResponse> {
             override fun onResponse(call: Call<CharactersResponse>, response: Response<CharactersResponse>) {
                 if (response.isSuccessful) {
-                    if (response.body() != null) {
-                        titleToChange = findViewById(R.id.title_to_change)
-                        titleToChange.text = response.body()?.toString()
-                        Log.d("Successful response", response.body().toString())
-                    }
-                    Log.d("Successful response", response.body().toString())
+                    val firstCharacter = response.body()?.results?.firstOrNull()
+                    firstCharacter?.let {
+                        characterNameTitle = findViewById(R.id.character_name)
+                        val characterNameData = getString(R.string.character_name) + it.name
+                        characterNameTitle?.text = characterNameData
+
+                        characterStatusTitle = findViewById(R.id.character_status)
+                        val characterStatusData = getString(R.string.character_status) + it.status
+                        characterStatusTitle?.text = characterStatusData
+
+                        Picasso.get().load(it.image).into(findViewById<ImageView>(R.id.character_image));
+                    } ?: Snackbar.make(findViewById(R.id.main), "No characters info", Snackbar.LENGTH_SHORT).show()
+
                 } else {
-                    Log.d("Not successful response", response.code().toString())
+                    Snackbar.make(findViewById(R.id.main), "No successful response" + response.code(), Snackbar.LENGTH_SHORT).show()
```

Comment:
Remember to not use literal strings "whatever" add them to strings.xml and you can obtain them here using context.getString(R.string.whatever)